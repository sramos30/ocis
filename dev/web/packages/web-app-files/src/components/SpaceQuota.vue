<template>
  <div class="space-quota">
    <p class="oc-mb-s oc-mt-rm" v-text="spaceStorageDetailsLabel" />
    <oc-progress
      :value="parseInt(quotaUsagePercent)"
      :max="100"
      size="small"
      :variation="quotaProgressVariant"
    />
  </div>
</template>

<script>
import filesize from 'filesize'

export default {
  name: 'SpaceQuota',
  props: {
    spaceQuota: {
      type: Object,
      required: true
    }
  },
  computed: {
    spaceStorageDetailsLabel() {
      if (this.spaceQuota.total) {
        return this.$gettextInterpolate(
          this.$gettext('%{used} of %{total} used (%{percentage}% used)'),
          {
            used: this.quotaUsed,
            total: this.quotaTotal,
            percentage: this.quotaUsagePercent
          }
        )
      }

      return this.$gettextInterpolate(this.$gettext('%{used} used (no restriction)'), {
        used: this.quotaUsed
      })
    },
    quotaTotal() {
      return filesize(this.spaceQuota.total)
    },
    quotaUsed() {
      return filesize(this.spaceQuota.used)
    },
    quotaUsagePercent() {
      return ((this.spaceQuota.used / this.spaceQuota.total) * 100).toFixed(1)
    },
    quotaProgressVariant() {
      switch (this.spaceQuota.state) {
        case 'normal':
          return 'primary'
        case 'nearing':
          return 'warning'
        case 'critical':
          return 'warning'
        default:
          return 'danger'
      }
    }
  }
}
</script>
