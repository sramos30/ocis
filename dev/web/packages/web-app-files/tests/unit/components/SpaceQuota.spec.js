import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import GetTextPlugin from 'vue-gettext'

import SpaceQuota from '@files/src/components/SpaceQuota'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(DesignSystem)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

describe('SpaceQuota component', () => {
  it('renders the space storage quota label', () => {
    const wrapper = getWrapper({ total: 10, used: 1, state: 'normal' })
    expect(wrapper.find('.space-quota').exists()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
  it.each([
    { state: 'normal', expectedVariation: 'primary' },
    { state: 'nearing', expectedVariation: 'warning' },
    { state: 'critical', expectedVariation: 'warning' },
    { state: 'exceeded', expectedVariation: 'danger' }
  ])('renders the progress variant correctly', (dataSet) => {
    const wrapper = getWrapper({ total: 10, used: 1, state: dataSet.state })
    const progressBar = wrapper.find('.space-quota oc-progress-stub')
    expect(progressBar.exists()).toBeTruthy()
    expect(progressBar.props().variation).toBe(dataSet.expectedVariation)
  })
})

function getWrapper(spaceQuota) {
  return shallowMount(SpaceQuota, {
    localVue,
    propsData: {
      spaceQuota
    }
  })
}
