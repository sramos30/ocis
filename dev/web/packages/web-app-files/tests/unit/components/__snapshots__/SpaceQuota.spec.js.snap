// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceQuota component renders the space storage quota label 1`] = `
<div class="space-quota">
  <p class="oc-mb-s oc-mt-rm">1 B of 10 B used (10.0% used)</p>
  <oc-progress-stub value="10" max="100" size="small" variation="primary"></oc-progress-stub>
</div>
`;
